sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -229,16 +232,20 @@ case class Uniform(min: Expression, max: Expression, seedExpression: Expression,
         if Seq(first, second).forall(integer) => IntegerType
       case (_, ShortType) | (ShortType, _)
         if Seq(first, second).forall(integer) => ShortType
+      case (_, ByteType) | (ByteType, _)
+        if Seq(first, second).forall(integer) => ByteType
       case (_, DoubleType) | (DoubleType, _) => DoubleType
       case (_, FloatType) | (FloatType, _) => FloatType
+      case (_, d: DecimalType) => d
+      case (d: DecimalType, _) => d
```

Comment:
if it's two decimals, shouldn't we pick the wider one instead of preferring the right one?

Reply:
Good idea, this is done.